### PR TITLE
Vfork Support

### PR DESCRIPTION
**Pull request by karya0**
Vfork support is implemented by suspending all user threads until the vfork call has returned in the parent. This would happen after the child has either exited or exec'd into a new program.

Each plugin should also use the following vfork-related events:
* DMTCP_EVENT_VFORK_PREPARE: Save a copy of any local data that might get overwritten by subsequent syscalls from the child.
* DMTCP_EVENT_VFORK_PARENT/FAILED: Restore local data saved during the PREPARE phase.
* DMTCP_EVENT_VFORK_CHILD: To perform any child-specific operations (e.g., setup locks, etc.).

### CODE REVIEW

**Comment by gc00** [comment]
I'm seeing 56 commits, and no text describing the PR itself.  Could you add text about the general principles that led you to create 56 different commits?  Otherwise, this becomes very difficult to seriously review.  Thanks.

**Comment by karya0** [comment]
> I'm seeing 56 commits, and no text describing the PR itself. Could you add text about the general principles that led you to create 56 different commits? Otherwise, this becomes very difficult to seriously review. Thanks.

Hi @gc00, this PR is still work-in-progress. I'm waiting for Cadence folks to verify that it works in their environment. 

Having said that, this PR includes commits from several other PRs. Once those PRs are merged, I'll rebase this PR.

**Comment by gc00** [comment]
And for the other PRs, I'm just asking to add comments to the code.  I'll even write the comment myself, and give it to you for insertion into the code, if you like!  Then I can approve it.  My primary worry is to keep the code maintainable by future code readers who don't know the history.

**Comment by karya0** [comment]
@gc00: I'll add comments and update this PR once we review/merge https://github.com/dmtcp/dmtcp/pull/835. Does that sound reasonable?

**Comment by karya0** [comment]
@gc00 : This PR is now ready for review.

File: src/plugin/pid/pid_miscwrappers.cpp [hunk 1]
```diff
@@ -66,6 +72,82 @@ fork()
...
+static void postVforkRestoreStack()
+{
+}
+
+extern "C" pid_t
+vfork()
+{
+  char dummy = 0;
+  //
+  // TODO: Add more comments before committing the code.
```
Comment (gc00) [inline src/plugin/pid/pid_miscwrappers.cpp:99]: @karya0 , could you do what the comment says?  :-)
`// TODO: Add more comments before committing the code.`

**Pull request merged**